src/planscape/forsys/cluster_stands_test.py

```diff
@@ -0,0 +1,453 @@
+            pixel_dist_to_condition_values, 4, 2, 1, priority_weights,
+            7.5 * 1e-2 / 1.5, 3)
+        self.assertDictEqual(clustered_stands_piw.clusters_to_stands,
+                             {0: [(0, 0), (1, 0), (2, 0), (3, 0)],
+                              1: [(2, 1), (3, 1)],
+                              2: [(0, 1), (1, 1)]})
+
+        # Moving from 2 conditions to 3 conditions (i.e. scaling the condition
+        # features by 1.5): preference for a blob over a line (no change from
+        # baseline)
+        pixel_dist_to_condition_values_c3 = {
+            0: {0: {'foo': 0.5, 'bar': 0.5, 'baz': 0.5},
+                1: {'foo': 0.4, 'bar': 0.4, 'baz': 0.4}},
+            1: {0: {'foo': 0.5, 'bar': 0.5, 'baz': 0.5},
+                1: {'foo': 0.4, 'bar': 0.4, 'baz': 0.4}},
+            2: {0: {'foo': 0.5, 'bar': 0.5, 'baz': 0.5},
+                1: {'foo': 0.1, 'bar': 0.1, 'baz': 0.1}},
+            3: {0: {'foo': 0.5, 'bar': 0.5, 'baz': 0.5},
+                1: {'foo': 0.1, 'bar': 0.1, 'baz': 0.1}},
+        }
+        priority_weights_c3 = {
+            'foo': 10,
+            'bar': 5,
+            'baz': 7.5
+        }
+        clustered_stands_c3 = ClusteredStands(
+            pixel_dist_to_condition_values_c3, 4, 2, 1, priority_weights_c3, 1,
+            3)
+        self.assertDictEqual(clustered_stands_c3.clusters_to_stands,
+                             {0: [(0, 0), (0, 1), (1, 0), (1, 1)],
+                              1: [(2, 1), (3, 1)],
+                              2: [(2, 0), (3, 0)]})
+
+    # ----------------------------------------------------
+    # The following tests are for validating corner cases.
+    # ----------------------------------------------------
+
+    def test_handles_more_clusters_than_stands(self) -> None:
+        pixel_dist_to_condition_values = {
+            0: {0: {'foo': 0.5},  1: {'foo': 0.2}},
+            1: {0: {'foo': 0.45}, 1: {'foo': 0.2}},
+            2: {0: {'foo': 0.3},  1: {'foo': 0.6}},
+        }
+        priority_weights = {
+            'foo': 10
+        }
+        clustered_stands = ClusteredStands(pixel_dist_to_condition_values,
+                                           3, 2, 1, priority_weights, 0, 10)
+        self.assertEqual(clustered_stands.cluster_status_message,
+                         "num desired clusters >= num stands")
+
+    def test_raises_connectivity_error(self) -> None:
+        pixel_dist_to_condition_values = {
+            0: {0: {'foo': 0.5},  1: {'foo': 0.2}},
+            2: {0: {'foo': 0.3},  1: {'foo': 0.6}},
+        }
+        priority_weights = {
+            'foo': 10
+        }
+        with self.assertRaises(Exception) as context:
+            ClusteredStands(pixel_dist_to_condition_values,
+                            3, 2, 1, priority_weights, 0, 1)
+        self.assertEqual(
+            str(context.exception),
+            "the graph has 2 connected components - it's impossible to " +
```

Comment:
can we give this error in a more user friendly format? if we're surfacing this to the user, it might be difficult for them to understand

Reply:
Done.

The likelihood of this error coming up is very low for the number of clusters specified.